### ISSUE TITLE
Socket Closed error

### ISSUE THREAD

**Issue by kvivekkr** [qlik-demo-team/qdt-components#232]
I have been using qdt-components for quite some time and it has been working fine till now, except since last week I am getting a socket closed error which details as below:
"Uncaught (in promise) Error: Socket closed at createEnigmaError (enigma.js:384) at RPC.onClose (enigma.js:1523)". Qliksense Enterprise server I have is 13.9.6 February 2019 Patch 3.
Can you please suggest what might be the problem and where should I look at? My code base has not changed a bit by the way.

**Comment by yianni-ververis** [comment]
We have not updated 2.x for a while. Did something change on your servers? 

Go to Chrome Network tab and inspect the frames on the particular ws connection

**Comment by kvivekkr** [comment]
Thank you for the response.
Even I have not changed the code since the working version. As per your suggestion, I looked under Messages into the Networks tab, this is what I found.
Request made by client application:
{
	delta: true,
	handle: - 1,
	id: 1,
	jsonrpc: "2.0",
	method:	"OpenDoc",
	params: ["app I'd"]
} 

Response from the server:
{
	jsonprc: "2.0",
	method: "OnAuthenticationInformation",
	params: {
		loginUri: "http://servername:80/internal_windows_authentication/?targetId=671c5699-c49f-43e6-ac0c-312345d3439f6",
mustAuthenticate: true
		} 
} 

Usually in the response from Qlik Server the parameter mustAuthenticate should have value as false and the parameters should contain User Directory and User ID values. But it's not happening for some reason. Any idea, what could be causing this?

**Comment by VaMusaka** [comment]
I have also seen the same issue. I suspect that the document API is failing to authenticate as there are no set cookies on the client because the recent updates for Google Chrome updated the SameSite cookie enforcement. [Chromium Same Site Update ](https://www.chromium.org/updates/same-site) 

I updated my browser to allow all cookies from [Google Chrome Cookie Settings](chrome://settings/cookies?search=third) to resolve the issue. 

This, however, does not resolve the issue for all client browsers. where users can not change the settings of the browser. How would l resolve this issue?

**Comment by kvivekkr** [comment]
Indeed looks like a problem caused by recent Chrome update. Since this is centrally controlled in my organisation, can't do much about it. We are trying to find a workaround. I tried to set allow cookies for all websites in chrome and the issue is still there. So, we thought of using Firefox as client browser, and now the socket closed error is gone and I am able to render few of the Qdt components such as filter, but components using the useHypercube hooks such as QdtTable are throwing "too much recursion" error. The error is particularly at or around line number 143 in useHypercube.js. Let me know how this can be handled.

**Comment by VaMusaka** [comment]
UPDATE ON MY PREVIOUS POST

After some more investigation I found this post [Qlik Sense Missing SameSite attribute...](https://community.qlik.com/t5/Qlik-Support-Knowledge-Base/Qlik-Sense-Missing-SameSite-attribute-now-blocks-requests-in/ta-p/1712551?_ga=2.267665870.1108272242.1597143678-356768904.1597143677) and updated my virtual proxy for ticket authentication to the recommended and now looks like the image below 

![image](https://user-images.githubusercontent.com/6652780/89989464-e5ee3000-dc78-11ea-8682-2b24fbf230a6.png)

this resolved the cookies issue for me in google Chrome. 

I have not used the hypercube component yet unfortunately l will try is and let you know.

**Comment by yianni-ververis** [comment]
Also I believe there a was a fix for that in the April release. Can you update your Qlik Sense servers to the latest?

**Comment by kvivekkr** [comment]
Sorry for responding late to this. So I went through the link that @VaMusaka shared and instead of going for server side changes, I set the flag "SameSite by Default Cookies" to Disabled and it started working. Thanks a lot guys for the help. It was Chrome all along.
For anyone else facing this recently, type chrome://flags/#same-site-by-default-cookies and set its value to Disabled from the drop down. If that doesn't work, then go for server side changes as described in the post shared by @VaMusaka.
